### PR TITLE
[Routing] Add matched and default parameters to redirect responses

### PR DESCRIPTION
**Pull request by artursvonda**
| Q | A |
| --- | --- |
| Branch | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php [hunk 1]
```diff
@@ -59,24 +62,28 @@ public function match($pathinfo)
...
-                    return array('_route' => 'baz');
+                    $ret = array('_route' => 'baz');
+                    return $ret;
                 }
 
                 // baz2
                 if ($pathinfo === '/test/baz.html') {
-                    return array('_route' => 'baz2');
+                    $ret = array('_route' => 'baz2');
+                    return $ret;
```
Comment (HeahDude) [inline src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php]: Are those changes intended ? You should revert them.

Comment (fabpot) [inline src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php]: This is a fixture file and the code is generated, this is intended.

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
Original PR is #11380

File: src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php (hunk 1 above)
Comment (HeahDude) [inline src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php]: Thanks, I've missed that.

**Comment by xabbuh** [comment]
Is this now considered a bug fix or a new feature (I seem to miss the conclusion in #11380)?

**Comment by fabpot** [comment]
Indeed, I messed up in the other PR. @Tobion was saying that this should be considered as a new feature. Anyway, I can still merge this one in master instead of 2.3. Sorry for the confusion.

@Tobion Does it look good?

**Comment by fabpot** [comment]
@Tobion Can you give us your opinion on this PR?

**Comment by Tobion** [comment]
I finished it in #19037. Rebased against master as it didn't apply cleanly and fixed adding the route params for the slash redirect as well. This was only done in the dumped version but not in the RedirectableUrlMatcher directly.

**Pull request closed without merging**